bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: admin endpoints sso migration

## Description
KFM changes (without connector related changes) for admin API endpoints authorization migration to internal RH SSO

## Verification Steps
1. Add a cluster (can be a non-existent one) to the `config/dataplane-cluster-configuration.yaml`, so that kafka can be created in the database
2. Change the `config/admin-authz-configuration.yaml` so that a rover group that you are member of is present in the configuration (this will enable you accessing admin API endpoints)
3. Create a kafka (it will be in failed state without using existing OSD cluster, it doesn't matter)
4. login to the db container (`make db/login`) and change `organisation_id` to one that you are not a member of, e.g. 

`update kafka_requests set organisation_id = '12147054';`

5. use rhoas-cli to get access token:

- Download https://github.com/redhat-developer/app-services-cli
- `mkdir ~/rhoas-config`
- `export RHOASCONFIG=~/rhoas-config/config.json`
- `export AUTHURL=https://auth.redhat.com/auth/realms/EmployeeIDP`
- `rhoas login --auth-url=$AUTHURL`
- `rhoas authtoken`

6. Confirm that access to the admin API endpoints is granted by hitting `http://localhost:8000/api/kafkas_mgmt/v1/admin/kafkas` with the token obtained in step 5.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment) - partially (connectors tests are broken)
